### PR TITLE
Diego/ora 1215 topichandler not updating state

### PR DESCRIPTION
**Pull request by xmariachi**
* Move change of state from topicsHandler to EndBlocker (otherwise it doesnt get persisted)
* e2e worker full cycle tested (incl unfulfilled & fulfilling nonces)

### CODE REVIEW

File: x/emissions/module/abci.go [hunk 1]
```diff
@@ -90,5 +91,32 @@ func EndBlocker(ctx context.Context, am AppModule) error {
...
+						if err != nil {
+							fmt.Println("Error adding worker nonce: ", err)
+							return
+						}
+					}
+				}(topic)
+			}
+		}
+	}
+	wgAddingNonces.Wait()
```
Comment (guilherme-brandao) [inline x/emissions/module/abci.go:119]: there's no need to query churnReadyTopics again + iterate through then. You already have everything available above in `topTopicsActiveWithDemand`. I guess you just need add the `AddWorkerNonce` right below `UpdateTopicEpochLastEnded`.

**Review by guilherme-brandao — CHANGES_REQUESTED** [review]

File: x/emissions/module/abci.go (hunk 1 above)
Comment (xmariachi) [reply x/emissions/module/abci.go:119]: Good catch, changing it now.

File: integration/postinit.sh [hunk 2]
```diff
@@ -23,5 +23,6 @@ mv $GENESIS.tmp $GENESIS
 
 
 echo "Starting allorad daemon and sleep for 3 seconds to let it start"
-$ALLORAD_BIN start & disown;
-sleep 3
+# $ALLORAD_BIN start & disown;
+# sleep 3
+$ALLORAD_BIN start
```
Comment (relyt29) [inline integration/postinit.sh:28]: why are you starting this without disowning, also the sleeping affects stability as well

Comment (xmariachi) [reply integration/postinit.sh:28]: sorry - that slipped in. I used that for my local testing without it. I'll change it. good catch.

Comment (xmariachi) [reply integration/postinit.sh]: done

**Review by kpeluso — APPROVED** [review]

**Pull request merged**